### ISSUE TITLE
ikhal crashes when deleting single instance of reccuring event

### ISSUE THREAD

**Issue by hamdav** [pimutils/khal#1207]
I've created a reccuring event happening every friday, six times. Deleting the last occurance (marking it and pressing d, selecting "Only this", and then pressing q to quit) causes the error below. I am using version 0.10.5 with python 3.10.8 on arch linux. When opening ikhal again, the event is not deleted. If I choose to delete all occurances, there is no problem. 

```
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/khal/ui/__init__.py", line 1365, in start_pane
    loop.run()
  File "/usr/lib/python3.10/site-packages/urwid/main_loop.py", line 287, in run
    self._run()
  File "/usr/lib/python3.10/site-packages/urwid/main_loop.py", line 385, in _run
    self.event_loop.run()
  File "/usr/lib/python3.10/site-packages/urwid/main_loop.py", line 790, in run
    self._loop()
  File "/usr/lib/python3.10/site-packages/urwid/main_loop.py", line 827, in _loop
    self._watch_files[fd]()
  File "/usr/lib/python3.10/site-packages/urwid/raw_display.py", line 416, in <lambda>
    wrapper = lambda: self.parse_input(
  File "/usr/lib/python3.10/site-packages/urwid/raw_display.py", line 515, in parse_input
    callback(processed, processed_codes)
  File "/usr/lib/python3.10/site-packages/urwid/main_loop.py", line 412, in _update
    self.process_input(keys)
  File "/usr/lib/python3.10/site-packages/urwid/main_loop.py", line 519, in process_input
    something_handled |= bool(self.unhandled_input(k))
  File "/usr/lib/python3.10/site-packages/urwid/main_loop.py", line 565, in unhandled_input
    return self._unhandled_input(input)
  File "/usr/lib/python3.10/site-packages/khal/ui/base.py", line 205, in on_key_press
    self.backtrack()
  File "/usr/lib/python3.10/site-packages/khal/ui/base.py", line 190, in backtrack
    cb(data)
  File "/usr/lib/python3.10/site-packages/khal/ui/__init__.py", line 1119, in cleanup
    self.collection.update(event)
  File "/usr/lib/python3.10/site-packages/khal/khalendar/khalendar.py", line 170, in update
    assert event.etag
AssertionError
```

**Comment by ei14** [comment]
Confirming that this is not a fluke. I also have Khal 0.10.5 and Python 3.10.8, and I experienced the exact same issue. I am using Arch Linux 6.0.11-arch1-1.

**Comment by TunnelThruTime** [comment]
can confirm this error as well. using os x 10.15.7, khal 0.10.5, python 3.8

**Comment by geier** [comment]
I believe this was fixed with #1158 which was released with v0.11.0. Please re-open if this is still an issue with v0.11.0 or later versions.
